src/library/trackcollection.cpp

```diff
@@ -510,4 +514,80 @@ StorageUnicodeFunctions::likeFunction(sqlite3_context *p,
                                     itrString, endString, E));
 }
 */
+//Private function
+
+bool TrackCollection::settingsPathStatus(){
+    QFileInfo fle(m_pConfig->getSettingsPath());
+    QFile::Permissions permit = fle.permissions();
+    permit = permit & MASK;
+    const short int NONE = QFile::NoOptions;
+    const short int EXE = QFile::ExeOwner;
+    const short int WRITE = QFile::WriteOwner;
+    const short int WRITE_EXE = QFile::WriteOwner | QFile::ExeOwner;
+    const short int READ = QFile::ReadOwner;
+    const short int READ_EXE = QFile::ReadOwner | QFile::ExeOwner;
+    const short int READ_WRITE = QFile::ReadOwner | QFile::WriteOwner;
+    switch(permit)
+    {
+        case NONE:
+            QMessageBox::critical(0, tr("Cannot Reach the path"),
+                                     tr("The Settings Path that you have "
+                                         "provided seems not to accessible, "
+                                         "because read, write and execute " 
+                                         "permissions to the path not given\n"
+                                         "Click OK to exit."), QMessageBox::Ok);
+                                return false;
+        case EXE:
+            QMessageBox::critical(0, tr("Cannot read and write to the given Path"),
+                                     tr("The Settings Path that you have "
+                                         "provided does not have the "
+                                         "permission to read and write the "
+                                         "files in it, so essential files "
+                                         "can't be read and log files can't "
+                                         "be written\n"
+                                         "Click OK to exit."), QMessageBox::Ok);
+                                return false;
+        case WRITE:
+            QMessageBox::critical(0, tr("Cannot read and access the directory"),
+                                     tr("The Settings Path that you have "
+                                         "provided does not have the "
+                                         "permission to access in it, so "
+                                         "can't process further.\n"
+                                         "Click OK to exit."), QMessageBox::Ok);
+                                return false;
+        case WRITE_EXE:
+            QMessageBox::critical(0, tr("Cannot read in the directory"),
```

Comment:
How about this: 

tr("Cannot read at settingsPath"),
tr("%1 does not have the " 
"permission to read, so can't read "
"the necessary files in it\n"
"Click OK to exit.").arg(m_pConfig->getSettingsPath())